# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../gcc/i386-Os.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Do not silently alter the ABI, even GCC/crt set_fast_math crashes otherwise.

  - Rene Rebe <rene@exactcode.de>

diff -urN gcc-4.1.1/gcc/config/i386/i386.c gcc-4.1.1-orig/gcc/config/i386/i386.c
--- gcc-4.1.1/gcc/config/i386/i386.c	2006-05-17 03:11:59.000000000 +0200
+++ gcc-4.1.1-orig/gcc/config/i386/i386.c	2007-02-15 17:45:47.037729849 +0100
@@ -1505,9 +1505,7 @@
      The default of 128 bits is for Pentium III's SSE __m128, but we
      don't want additional code to keep the stack aligned when
      optimizing for code size.  */
-  ix86_preferred_stack_boundary = (optimize_size
-				   ? TARGET_64BIT ? 128 : 32
-				   : 128);
+  ix86_preferred_stack_boundary = 128;
   if (ix86_preferred_stack_boundary_string)
     {
       i = atoi (ix86_preferred_stack_boundary_string);
